query.go

```diff
@@ -284,7 +284,14 @@ func (rows *Rows) Scan(dest ...interface{}) (err error) {
 				rows.Fatal(err)
 			}
 		default:
-			rows.Fatal(fmt.Errorf("Scan cannot decode into %T", d))
+			switch vr.Type().DataType {
```

Comment:
Is there a better way to do this? It seems hacky but I couldn't find a reasonable way to do it other than matching on interface{} and then falling through into the default case above (which was even more hacky).